### ISSUE TITLE
Support for Properties argument in nuget pack

### ISSUE THREAD

**Issue by daveaglick** [cake-build/cake#666]
When packing a project file (#637), the `-Properties` argument is often used to specify the build configuration (or perform substitution in an adjacent nuspec):

```
nuget pack foo.csproj -Build -Symbols -Properties Configuration=Release
```

Support for specifying multiple properties should be added to `NuGetPackSettings` and `NuGetPacker.GetArguments()`. The generated argument should separate multiple properties with a semicolon per the NuGet command line reference.

**Comment by daveaglick** [comment]
Code is done, but not tests. Holding off on tests and submitting a PR until #662 and #667 are merged to avoid submitting a PR based on a PR based on a PR. The tests for this one will be based on project packing tests from #667 since the `-Properties` argument is only used for packing project files.

**Comment by gep13** [comment]
:+1: just thinking about this one as well, so that an alternative is mentioned...

You could use the ArgumentCustomization method on the base ToolSettings class to extend the arguments that are supported by an alias.

**Comment by daveaglick** [comment]
@gep13 You mean as a user (I.e., if I wanted to get the `Properties` argument into the settings today without any code changes)? Would that work since `NuGetPacker.GetArguments()` doesn't ever evaluate `ToolSettings.ArgumentCustomization`? Perhaps the following should also be added to `NuGetPacker.GetArguments()`:

```
if (settings.ArgumentCustomization != null)
{
    builder = settings.ArgumentCustomization(builder);
}
```

**Comment by gep13** [comment]
Assuming that NuGetTool derives from Tool, which I assume it does, then the base class has this covered:

https://github.com/cake-build/cake/blob/2e33d9f6c6c8a7c89d4f1a4c31ec70daf31f009e/src/Cake.Core/Tooling/Tool.cs#L134

No?

**Comment by gep13** [comment]
@patriksvensson did you mean to close this issue?

**Comment by gep13** [comment]
I didn't mean to close it, that was an accident on my part, but then I clicked re-open, and immediately it got closed by you :smile:

**Comment by daveaglick** [comment]
Hahaha - "THIS IS THE WORST IDEA EVAR!!!! CLOSE!!!!"

**Comment by daveaglick** [comment]
@gep13 

> the base class has this covered

Indeed it does! I wasn't following the flow close enough and didn't notice that `NuGetPacker.GetArguments()` actually returns a `ProcessArgumentBuilder` that gets passed to the  process runner.

**Comment by daveaglick** [comment]
It's good to know about the ability to customize the settings arguments - that'll come in handy.

I still think the explicit Properties dictionary is warranted since it makes things a little clearer (especially with the slightly goofy `key=value;key=value` argument syntax).

**Comment by gep13** [comment]
>@daveaglick said...
I still think the explicit Properties dictionary is warranted since it makes things a little clearer

oh yeah, absolutely, I am not suggesting that this isn't needed.  I just wanted to make sure that you were aware of the option to do it using the other technique, in the interim.

**Comment by daveaglick** [comment]
:+1:

**Comment by patriksvensson** [comment]
@daveaglick @gep13 Seriously, I have no idea why this issue got closed. I've been at a dinner tonight and only looked at the issue on my phone when I got a notification...

**Comment by gep13** [comment]
@patriksvensson so did you re-open it as well?  If not, maybe a good time to change your GitHub password :cry:

**Comment by devlead** [comment]
Have you checked the issue number...👿

**Comment by patriksvensson** [comment]
@gep13 Yes, I reopened the issue manually when I saw your conversation and I have two factor auth.

**Comment by daveaglick** [comment]
So tempted to close and immediately reopen this again just to mess with everyone...

**Comment by patriksvensson** [comment]
@daveaglick lol :smile:

**Comment by Philo** [comment]
@patriksvensson, any chance this could be included in milestone v0.9.0 also.  This is closely related to #637 in terms of features that will be utilised together.

For example, I'm looking to iterate through projects in a solution containing nuspec files, then do a "Nuget pack" against the project file and pass the configuration as a property to nuget to determine the files to pack:

nuget pack MyProject.csproj -Version 1.0.0 -Properties Configuration=Release -OutputDirectory ./artifacts -IncludeReferencedProjects
